packages/components/src/sites-table/site-status.tsx

```diff
@@ -0,0 +1,40 @@
+import { useI18n } from '@wordpress/react-i18n';
+
+export const siteStatuses = [ 'public', 'private', 'coming-soon' ] as const;
+
+export type SiteStatus = typeof siteStatuses[ number ];
+
+export interface SiteObjectWithStatus {
+	is_coming_soon?: boolean;
+	is_private?: boolean;
+	launch_status?: string;
+}
+
+export const getSiteStatus = ( site: SiteObjectWithStatus ): SiteStatus => {
+	if ( site.is_coming_soon || ( site.is_private && site.launch_status === 'unlaunched' ) ) {
+		return 'coming-soon';
+	}
+
+	if ( site.is_private ) {
+		return 'private';
+	}
+
+	return 'public';
+};
+
+export const getTranslatedStatuses = (
```

Comment:
This function feels a little awkward to me; having a function that accepts the `__()` function as an argument. I do it sometimes but try not to export them, I think they're like an internal detail.

Could this be a hook instead? Like `useTranslatedStatuses()`? It'd still be easy to use in `useSitesTableFiltering()`.